### ISSUE TITLE
No access to backup after reinstallation

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#7890]
**Describe the bug:**
For testing reasons I had to uninstall and reinstall between several builds. While doing this I stumbled upon this usability problem described as follows.

**To Reproduce:**
1. Have c:geo installed and working (account data, stored caches, etc.)
2. Perform back of the database and of full settings
3. Make sure backup files are stored in `storage/emulated/cgeo/backup` in your user storage
4. Deinstall c:geo
5. Reinstall c:geo (same version)
6. Interrupt "First steps" screen by pressin back
7. Accept file and position permission which are prompted then
8. Go to Settings - Backup/Restore

**Actual behavior/state after performing these steps:**
The menu shows that neither a database backup nor a settings backup is available.
Looking into "About c:geo" - "System" I can see that c:geo is using the fallback internal directory (`/data/user/0/cgeo.geocaching/`) as user storage directory.

Only after force stopping and restarting c:geo it seems to recognize the availability of the user storage dir at `storage/emulated/0/cgeo/`.
Subsequently I was able to restore the backups.

**Expected behavior/state after performing these steps:**
Restoring (and usage of the correct user storage directory) should be possible already in the first session as its most likely users would like use restore functions especially during the first session.

**Version of c:geo used:**
2019.10.08-NB

**Is the problem reproducible:**
Yes

@pstorch 
Any idea why fallback happens? AFAIR your implementation of choosing the appropriate directory was implemented before we added runtime permissions. So runtime permissions might interfere here somehow.

**Comment by moving-bits** [comment]
@pstorch Thanks for the helpful pointer!

I created PR #7891 to fix this issue + a followup issue in creating the prefs backup file.
